app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt

```diff
@@ -0,0 +1,166 @@
+                .setPositiveButton("Apply"){_,_ ->
+                    Repository.recipesList.remove(recipe)
+                    Repository.recipesListFilterForCategoryClick.remove(recipe)
+                    mainActivity.deleteRecipe(recipe.recipeName)
+                    addRecipeToListAndDataBase(binding)
+                    Repository.listOfRecipeForDetail.clear()
+                    Repository.listOfRecipeForDetail.add(newRecipeForDetails)
+                    listener()
+                }
+                .setNegativeButton("Cancel",null)
+                .create()
+    }
+
+    private fun showDialog(){
+        Repository.listOfSelectedIngredientsForRecipe.clear()
+        lateinit var dialog: AlertDialog
+        val arrayIngredients:Array<String> = Repository.IngredientsList.map { it.ingredientName }.toTypedArray()
+        val mutableListFalse = mutableListOf<Boolean>()
+
+        val recipe = Repository.recipeToEdit
+        val listOfIngredientsS = recipe.ingredientsToUse.split(",")
+        for(ingredient in Repository.IngredientsList){
+            if(listOfIngredientsS.contains(ingredient.ingredientName) || listOfIngredientsS.contains(ingredient.ingredientName+" ,") ||  listOfIngredientsS.contains(" "+ingredient.ingredientName)){
+                mutableListFalse.add(true)
+            }else {
+                mutableListFalse.add(false)
+            }
+        }
+
+        val arrayChecked:BooleanArray = mutableListFalse.map { it }.toBooleanArray()
+        val builder = AlertDialog.Builder(requireContext())
+        builder.setTitle("Choose your ingredients.")
+
+        builder.setMultiChoiceItems(arrayIngredients, arrayChecked) { dialog, which, isChecked->
+            arrayChecked[which] = isChecked
+        }
+
+        builder.setPositiveButton("Add") { _, _ ->
+            binding.editTextView.text = "Selected Ingredients \n"
+            for (i in arrayIngredients.indices) {
+                val checked = arrayChecked?.get(i)
+                if (checked!!) {
+                    binding.editTextView.text = "${binding.editTextView.text} ${arrayIngredients[i]},"
+                    Repository.listOfSelectedIngredientsForRecipe.add(arrayIngredients[i])
+                }
+            }
+        }
+
+        builder.setNegativeButton("Cancel",null)
+
+        dialog = builder.create()
+        dialog.show()
+    }
+
+    fun addRecipeToListAndDataBase(binding: FragmentEditRecipeBinding){
+        var ingredientsSelectedString = ""
+        for(ingredient in Repository.listOfSelectedIngredientsForRecipe){
+            ingredientsSelectedString += "$ingredient,"
+        }
+        ingredientsSelectedString = ingredientsSelectedString.dropLast(1)
+        val newRecipe = Recipes(
+                recipeName = binding.editRecipeNameEditText.text?.toString()?:"",
+                ingredientsToUse = if(ingredientsSelectedString ==""){
+                    binding.editTextView.text.toString()
+                }else{
```

Comment:
spacing.  just some examples, but go through and clean it up.  maybe use find/replace to speed it up on some of the larger files.  

spacing is important because it helps the eye see the code in a regular way and helps you find your bugs, makes it easier to find logical bugs also.



```suggestion
                } else {
```